File: kotlin-spring-data-jpa/src/test/kotlin/com/kotlin_spring_data_jpa/kotlin_spring_data_jpa/repositories/CustomerRepositoryTest.java
package com.kotlin_spring_data_jpa.kotlin_spring_data_jpa.repositories;

import com.kotlin_spring_data_jpa.kotlin_spring_data_jpa.entities.Customer;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;

import static org.assertj.core.api.Assertions.assertThat;

@DataJpaTest
public class CustomerRepositoryTest {
    @Autowired
    private CustomerRepository customerRepository;

    @Test
    void testSaveCustomer() {
        Customer customer = new Customer();
        customer.setName("Test Customer");

        Customer savedCustomer = customerRepository.save(customer);
        assertThat(savedCustomer).isNotNull();
        assertThat(savedCustomer.getId()).isNotNull();
        assertThat(savedCustomer.getName()).isEqualTo(customer.getName());
    }
}

File: kotlin-spring-r2dbc/src/test/kotlin/com/kotlin_spring_r2dbc/kotlin_spring_r2dbc/repositories/CustomerRepositoryTest.kt
package com.kotlin_spring_r2dbc.kotlin_spring_r2dbc.repositories

import com.fasterxml.jackson.databind.ObjectMapper
import com.kotlin_spring_r2dbc.kotlin_spring_r2dbc.config.DatabaseConfig
import com.kotlin_spring_r2dbc.kotlin_spring_r2dbc.domain.Customer
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.autoconfigure.data.r2dbc.DataR2dbcTest
import org.springframework.context.annotation.Import

@DataR2dbcTest
@Import(DatabaseConfig::class)
class CustomerRepositoryTest {
    @Autowired
    lateinit var customerRepository: CustomerRepository

    @Test
    fun testCreateJson() {
        val objectMapper = ObjectMapper()

        println(
            objectMapper.writeValueAsString(
                getTestCustomer()
            )
        )
    }

    @Test
    fun saveNewBeer() {
        customerRepository.save(getTestCustomer())
            .subscribe { customer -> println(customer.toString()) }
    }

    fun getTestCustomer(): Customer =
        Customer(
            name = "Alex Johnson"
        )
}